kotlin/samples/todo-android/common/src/main/java/com/squareup/sample/todo/MasterDetailAware.kt

```diff
@@ -0,0 +1,77 @@
+/*
+ * Copyright 2019 Square Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.squareup.sample.todo
+
+/**
+ * Informs [MasterDetailAware] renderings whether they're children of a [MasterDetailScreen],
+ * and if so in what configuration.
+ */
+enum class MasterDetailConfig {
+  /**
+   * Parent is not a [MasterDetailScreen]. Expected to be used as the default value
+   * of [MasterDetailAware.masterDetailConfig].
```

Comment:
I don't know what this value actually implies if this is the config that is being rendered. Is this a valid configuration to render as? If not, I like the nullable route better because it lets the type system enforce whether this default value is allowed (for example, shouldn't be allowed as an argument to `makeAware`).

Reply:
It is valid to be rendered with `None`, that's what happens if your parent isn't `MasterDetailScreen`. But you're right, `makeAware` should never be called with `None`. That sounds compelling to me.